external-crates/move/crates/move-analyzer/trace-adapter/src/runtime.ts

```diff
@@ -198,45 +201,63 @@ export class Runtime extends EventEmitter {
         // skip all events until the corresponding CloseFrame event,
         // pop the top frame from the stack, and proceed to the next event
         while (true) {
-            this.eventIndex++;
-            if (this.eventIndex >= this.trace.events.length) {
+            if (this.step(/* next */ false, /* stopAtCloseFrame */ true, /* nextLineSkip */ true)) {
+                // trace viewing session finished
                 throw new Error("Cannot find corresponding CloseFrame event for function: " +
                     currentFrame.name);
             }
             currentEvent = this.trace.events[this.eventIndex];
-            if (currentEvent.type === 'CloseFrame' && currentEvent.id === currentFrame.id) {
-                break;
+            if (currentEvent.type === 'CloseFrame') {
+                const currentFrameID = currentFrame.id;
+                // `step` call finished at the CloseFrame event
+                // but did not process it so we need pop the frame here
+                this.frameStack.frames.pop();
+                if (currentEvent.id === currentFrameID) {
+                    break;
+                }
             }
         }
-        this.frameStack.frames.pop();
-        return this.step(false);
+
+        // Do not skip to same line when stepping out as this may lead
```

Comment:
This change is unrelated to the "continue" action implementation, but it's small enough that I decided to sneak it in as part of this PR....